notebooks/Project 1 - Algorithms.ipynb

```diff
@@ -208,86 +262,86 @@
-    "plt.plot(n_points, recur, label=\"Recursive\") \n",
-    "plt.ylim(ymin=0, ymax=0.01)\n",
-    "plt.legend()\n",
-    "plt.title(\"Time to Calculate nth Fibonacci\")\n",
-    "plt.show();    "
+    "    # requesting modern fibonacci (0,1,1,2,3,5...)\n",
+    "    if start == 0:\n",
+    "        penultimate_term = 0\n",
+    "        start_index_modifier = 3\n",
+    "    # requesting historic fibonacci (1,1,2,3,5...)\n",
+    "    else:\n",
+    "        penultimate_term = 1\n",
+    "        start_index_modifier = 2\n",
+    "\n",
+    "    for i in range(start + start_index_modifier, n + 1):\n",
+    "        next_term = ultimate_term + penultimate_term\n",
+    "        penultimate_term = ultimate_term\n",
+    "        ultimate_term = next_term\n",
+    "    return next_term\n",
+    "\n",
+    "\n",
+    "def recursive_fibonacci(n, start):\n",
+    "    current_term = 0\n",
+    "    # Conditionally differentiating modern and historic sequences\n",
+    "    start_index_modifier = 3 if start == 0 else 2\n",
+    "\n",
+    "    if n <= start_index_modifier:\n",
+    "        current_term = current_term + 1 + 0\n",
+    "        return current_term\n",
+    "    current_term = recursive_fibonacci(n - 1, start) + recursive_fibonacci(n - 2, start)\n",
+    "    return current_term\n",
+    "\n",
+    "\n",
+    "def fibonacci_compare(n, start, fib_type):\n",
+    "    \"\"\"Calculates Fibonacci sequence and appends calculation time to the fib_type key in time_dict\"\"\"\n",
+    "    start_timestamp = time.time()\n",
+    "\n",
+    "    if fib_type == 'Iterative':\n",
+    "        print(f\"Calculation type: {fib_type}\\n\"\n",
+    "              f\"Number of iterations: {n}\\n\"\n",
+    "              f\"Next number in Fibonacci sequence: {iterative_fibonacci(n, start)}\\n\")\n",
+    "    else:\n",
+    "        print(f\"Calculation type: {fib_type}\\n\"\n",
+    "              f\"Number of iterations: {n}\\n\"\n",
+    "              f\"Next number in Fibonacci sequence: {recursive_fibonacci(n, start)}\\n\")\n",
+    "    time_dict[fib_type].append(time.time() - start_timestamp)\n",
+    "\n",
+    "\n",
+    "# User input regarding starting number\n",
+    "start_number = -1\n",
+    "while (start_number != 1) and (start_number != 0):\n",
+    "    start_number = int(input(\"Do you want to start the series with 0 or 1? \"))\n",
+    "\n",
+    "# Arrays & maps for determining method parameters and plotting\n",
+    "n_points = list(range(3, 43, 5))\n",
+    "fibonacci_types = ['Iterative', 'Recursive']\n",
+    "time_dict = {k: [] for k in fibonacci_types}\n",
+    "\n",
+    "for n in n_points:\n",
+    "    for fibonacci_type in fibonacci_types:\n",
+    "        fibonacci_compare(n, start_number, fibonacci_type)\n",
+    "\n",
+    "# Plotting via iterating through time_dict and applying to subplot axes\n",
+    "f, axes = plt.subplots(2, sharex=True, figsize=(15,10))\n",
+    "for k, v in time_dict.items():\n",
```

Comment:
Iterates through the dictionary keys (`Iterative` and `Recursive`), and plots the `Recursive` times on one of the subplots, (`axes[0]`), and the `Iterative` times on the other subplot (`axes[1]`). When you have a figure with subplots, methods for setting things like `ylabel` and `title` are slightly different than a normal plot, but the meaning is the same. Also of note is the `xlabel` only has to be set once because the x axis is shared, and `suptitle` sets the title for the entire graph, rather than either subplot individually.